internal/file/file.go

```diff
@@ -62,82 +63,140 @@ func PackFiles(files []*os.File) (*os.File, int64, error) {
 	return tempFile, fileInfo.Size(), nil
 }
 
-// UnpackFiles gzip-decompresses and un-tars files into the current working directory
-// and returns the names and decompressed size of the created files
-func UnpackFiles(reader io.Reader, decideOverwrite OverwriteDecider) ([]string, int64, error) {
-	// chained readers -> gr reads from reader -> tr reads from gr
-	gr, err := pgzip.NewReader(reader)
+// ---------------------------------------------------- Unpack Files ---------------------------------------------------
+
+var ErrUnpackNoHeader = errors.New("no header in tar archive")
+var ErrUnpackFileExists = errors.New("file exists")
+var ErrUninitialized = errors.New("unpacker is uninitialized")
+
+// Unpacker defines a encapsulated unit for unpacking a compressed
+// tar archive
+type Unpacker struct {
+	prompt bool // prompt defines whether we should prompt the user to overwrite files
+	cwd    string
+
+	gr *pgzip.Reader
+	tr *tar.Reader
+	r  io.ReadCloser
+}
+
+func NewUnpacker(prompt bool) *Unpacker {
+	return &Unpacker{
+		prompt: prompt,
+	}
+}
+
+// Init initializes the unpacker with the provided reader.
+// NOTE: this has to be called **before** trying to use the unpacker.
+func (u *Unpacker) Init(r io.ReadCloser) error {
```

Comment:
Why not make this part of NewUnpacker instead? Makes it harder to misuse imo (but not impossible, since Unpacker is exported).